frontend/packages/console-app/src/components/user-preferences/UserPreferenceCustomField.tsx

```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+import { UserPreferenceCustomField as CustomFieldType } from '@console/dynamic-plugin-sdk/src';
+import { UserPreferenceFieldProps } from './types';
+
+type UserPreferenceCustomFieldProps = UserPreferenceFieldProps<CustomFieldType>;
+
+const UserPreferenceCustomField: React.FC<UserPreferenceCustomFieldProps> = ({
+  component: Component,
+  props,
+}) => (Component ? <Component {...props} /> : null);
```

Comment:
Only if you want make it more clear that props is a specific attribute here:

```suggestion
const UserPreferenceCustomField: React.FC<UserPreferenceCustomFieldProps> = ({
  component: Component,
  props: customComponentProps,
}) => (Component ? <Component {...customComponentProps} /> : null);
```

Reply:
If we are renaming it to `customComponentProps`, I guess we could also update from `Component` to `CustomComponent`.
WDYT?

Reply:
I like that idea.